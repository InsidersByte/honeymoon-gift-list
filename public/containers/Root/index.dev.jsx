/* @flow */

import React from 'react';
import App from '../App';
import DevTools from '../DevTools';

const styles = {
    root: {
        height: '100%',
        width: '100%',
    },
};

const Root = (props: Object) => (
    <div style={styles.root}>
        <App {...props} />
        <DevTools />
    </div>
);

export default Root;
