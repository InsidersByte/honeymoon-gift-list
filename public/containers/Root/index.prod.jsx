/* @flow */

import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import routes from '../../routes';

type PropsType = {
    store: Object,
    history: Object,
};

const Root = ({ store, history }: PropsType) => (
    <Provider store={store}>
        <Router history={history}>
            {routes(store)}
        </Router>
    </Provider>
);

export default Root;
