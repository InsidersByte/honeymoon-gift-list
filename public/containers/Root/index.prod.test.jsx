import React from 'react';
import { shallow } from 'enzyme';
import Root from './index.prod';
import configureStore from '../../store/configureStore.dev';

describe('Root', () => {
    it('should render correctly', () => {
        const store = configureStore();

        const wrapper = shallow(
            <Root store={store} history={{}}>
                <h1>Hello World</h1>
            </Root>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
