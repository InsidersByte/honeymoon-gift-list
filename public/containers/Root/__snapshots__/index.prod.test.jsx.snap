// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "liftedStore": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Route
      component={[Function]}
    />
  </BrowserRouter>
</Provider>
`;
